python/legate_dataframe/experimental_ibis/__init__.py

```diff
@@ -0,0 +1,309 @@
+            com.IbisError(
+                "Passing `database` to the Legate backend's `create_table()` method "
+                "is not supported: Legate cannot set a database."
+            )
+
+        if name in self._tables and not overwrite:
+            raise com.IbisError(f"Cannot overwrite existing table `{name}`")
+
+        if obj is None and schema is None:
+            raise com.IbisError("The schema or obj parameter is required")
+
+        if schema is not None:
+            schema = ibis.schema(schema)
+
+        if obj is not None:
+            table = self._convert_object(obj)
+            if schema is None:
+                schema = infer_schema_from_logical_table(table)
+            else:
+                # Check if schema matches actual table (is it right to not
+                # enforce the schema by dropping cols or casting?)
+                actual_schema = infer_schema_from_logical_table(table)
+                if schema != actual_schema:
+                    # TODO: General Q: when to use Ibis errors and when not?!
+                    # (I find the broad use of IbisError a bit surprising.)
+                    raise com.IbisError(
+                        f"Schema {schema} and inferred one {actual_schema} do not match."
+                    )
+        else:
+            raise NotImplementedError("Cannot yet create empty table from schema.")
+
+        self._tables[name] = table, schema
+
+        return self.table(name)
+
+    def create_view(
+        self,
+        name: str,
+        obj: ir.Table,
+        *,
+        database: str | None = None,
+        overwrite: bool = False,
+    ) -> ir.Table:
+        return self.create_table(
+            name, obj=obj, temp=None, database=database, overwrite=overwrite
+        )
+
+    def drop_table(self, name: str, *, force: bool = False) -> None:
+        if self._tables.pop(name, None) is None and not force:
+            raise com.IbisError(f"Table {name} does not exist") from None
+
+    def drop_view(self, name: str, *, force: bool = False) -> None:
+        self.drop_table(name, force=force)
+
+    def _convert_object(self, obj: Any) -> Any:
+        return _convert_object(obj, self)
+
+    @classmethod
+    @lru_cache
+    def _get_operations(cls):
+        return tuple(op for op in _execute_node.registry if issubclass(op, ops.Value))
+
+    @classmethod
+    def has_operation(cls, operation: type[ops.Value]) -> bool:
+        op_classes = cls._get_operations()
+        return operation in op_classes or issubclass(operation, op_classes)
+
+    def _to_ldf_table(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Expr, object] | None = None,
+        limit: int | Literal["default"] | None = None,
+        **kwargs: Any,
+    ) -> LogicalTable:
+        """Execute after ensuring that the result is a logical table."""
+        if limit is not None and limit != "default":
+            raise NotImplementedError("limit parameter not supported.")
```

Comment:
When Ibis is used interactively, we run all commands eagerly by executing them with an implicit `LIMIT 11` at the end, so we can return the first 10 rows of everything (and an indicator ellipsis if there are >10 rows).

From my understanding of legate execution, this model doesn't really work, so the missing `limit` won't really be an issue.

Reply:
Ah OK, somehow I thought I saw an error for it.  I guess that means just ignoring `limit` for now, thanks!

Reply:
For reference, I guess I got it from here: https://github.com/ibis-project/ibis/blob/70de8db00b83dd3652a0e1e123711ee80b68983e/ibis/backends/pandas/__init__.py#L314

Maybe it would make sense to at least support it by slicing the final (full) result.
(Although, the nice way would be to implement a/the `ops.Limit` which then could be optimized to be pushed down as far as possible as a re-write.